frontend/src/components/mine/MineHeader.js

```diff
@@ -138,13 +151,91 @@ class MineHeader extends Component {
             {ModalContent.ADD_TAILINGS}
           </button>
         </Menu.Item>
+
+        <AuthorizationWrapper inTesting>
+          {(!this.props.mine.verified_status || !this.props.mine.verified_status.healthy) && (
+            <Menu.Item key="2">
+              <Popconfirm
+                placement="left"
+                title="Are you sure?"
+                onConfirm={() => {
+                  this.props.setMineVerifiedStatus(this.props.mine.guid, true).then(() => {
+                    this.props.fetchMineRecordById(this.props.mine.guid);
+                    this.props.fetchMineVerifiedStatuses(
+                      `idir\\${this.props.userInfo.preferred_username}`
+                    );
+                  });
+                }}
+                okText="Yes"
+                cancelText="No"
+              >
+                <button type="button" className="full" onClick={(e) => e.stopPropagation()}>
+                  <img
+                    alt="checkmark"
+                    className="padding-small"
+                    src={SUCCESS_CHECKMARK}
+                    width="30"
+                  />
+                  Verify Mine Data
+                </button>
+              </Popconfirm>
+            </Menu.Item>
+          )}
+        </AuthorizationWrapper>
+        <AuthorizationWrapper inTesting>
+          {(!this.props.mine.verified_status || this.props.mine.verified_status.healthy) && (
+            <Menu.Item key="3">
+              <Popconfirm
+                placement="left"
+                title="Are you sure?"
+                onConfirm={() =>
+                  this.props.setMineVerifiedStatus(this.props.mine.guid, false).then(() => {
```

Comment:
To simplfy the jsx, move this logic into its own function and pass it into the `onConfirm`
you can also pass the event in here, and handle the propagation inside the new function. Then the `onClick` on the button below will no longer be needed.

Reply:
fixed